src/platform/packages/shared/kbn-scout/src/common/services/saml_auth.ts

```diff
@@ -42,16 +42,20 @@ const createKibanaHostOptions = (config: ScoutTestConfig): HostOptions => {
 
 export const createSamlSessionManager = (
   config: ScoutTestConfig,
-  log: ScoutLogger
-): SamlSessionManager => {
+  log: ScoutLogger,
+  workerIndex?: number
+) => {
   const resourceDirPath = getResourceDirPath(config);
   const rolesDefinitionPath = path.resolve(resourceDirPath, 'roles.yml');
 
   const supportedRoleDescriptors = readRolesDescriptorsFromResource(rolesDefinitionPath) as Record<
     string,
     unknown
   >;
-  const supportedRoles = Object.keys(supportedRoleDescriptors);
+
+  const customRoleName = workerIndex ? `custom_role_${workerIndex + 1}` : 'custom_role';
```

Comment:
to run tests with custom role in parallel, we will a unique role per worker to avoid overrides